README.md

```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
+        imagePullPolicy = ImagePullPolicy.Always //4
+)
 ```
-We need to specify a port in order to prevent dekorate it is about a Web Application, this way a service will be generated in the k8s manifest.
+- We need to specify a **`@Port`** in order to prevent dekorate it is about a Web Application, this way a service will be generated in the k8s manifest.
 
-NOTE: if you are using minikube a trick that often works is to use a service of type NodePort.
+- The host under which the application is going to be exposed.
+- **`expose`** controls whether the application should be exposed via Ingress.
+- We use **`Always`** in order to be able to use an updated image.
 
-Generate the manifests launching a project compilation. Navigate to the directory and run `mvn clean package`. The generated manifests can be found under `target/classes/META-INF/dekorate.
+Generate the manifests launching a project compilation. Navigate to the directory and run `mvn clean package`. The generated manifests can be found under `target/classes/META-INF/dekorate`.
 
-NOTE: To perform the following steps you need to be connected to a running kubernetes cluster.
+## Building and deploying
 
-Run the following command to create the resources defined in the manifest.
+We need to somehow build and push the image to the registry to make it available for the kubernetes cluster. 
+
+### Using docker
+A basic Dockerfile is provided in the project base directory so you can build the image using the following command:
+```
+docker build -f Dockerfile -t USERNAME/hello-world-fwless:1.0-SNAPSHOT .
+```
+Then, tag the image as follows to be able to push it to the local registry:
+```
+docker tag USERNAME/hello-world-fwless:1.0-SNAPSHOT localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
-kubectl apply -f target/classes/META-INF/dekorate/kubernetes.yml
+Finally, you can push the image to the local registry
+```
+docker push localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
 
-At last, we need to expose the service in order to make it accessible from outside the cluster. For that, list the service using the command:
+As we have pushed the image to the local registry, we need to modify manually the kubernetes yml file generated to use that image.
+So, replace the following line: 
+```
+image: USERNAME/hello-world-fwless:1.0-SNAPSHOT
+```
+by
 ```
-oc get services
+image: localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
 
-Copy the name of the service and expose it running:
+Finally, we will deploy the application by posting the manifest with the following command:
 
 ```
-oc expose service hello-world-fwless
+kubectl create ns demo
+kubectl apply -f target/classes/META-INF/dekorate/kubernetes.yml -n demo
 ```
 
-Now, you should be able to get the address of the service and access the application with a browser. To get the adress, tape:
+Check the application by opening a browser to http://fw-app.127.0.0.1.nip.io/api/hello or curl http://fw-app.127.0.0.1.nip.io/api/hello
```

Comment:
After a few seconds, check if the application is running and available at the following address opened within your browser `` or do a curl/wget or httpie within a terminal